SignalMessaging/utils/ConversationSearcher.swift

```diff
@@ -5,28 +5,61 @@
 import Foundation
 import SignalServiceKit
 
-public class ConversationSearchResult {
+public class ConversationSearchResult: NSObject, Comparable {
     public let thread: ThreadViewModel
 
     public let messageId: String?
 
     public let snippet: String?
 
-    init(thread: ThreadViewModel, messageId: String?, snippet: String?) {
+    private let sortKey: UInt64
+
+    init(thread: ThreadViewModel, messageId: String?, snippet: String?, sortKey: UInt64) {
         self.thread = thread
         self.messageId = messageId
         self.snippet = snippet
+        self.sortKey = sortKey
+    }
+
+    // Mark: Comparable
+
+    public static func < (lhs: ConversationSearchResult, rhs: ConversationSearchResult) -> Bool {
+        return lhs.sortKey < rhs.sortKey
+    }
+
+    // MARK: Equatable
+
+    public static func == (lhs: ConversationSearchResult, rhs: ConversationSearchResult) -> Bool {
+        if lhs.thread.threadRecord.uniqueId != rhs.thread.threadRecord.uniqueId {
+            return false
+        }
+        return NSObject.isNullableObject(lhs.messageId as NSObject?, equalTo: rhs.messageId as NSObject?)
```

Comment:
The type signature asserts that neither lhs nor rhs are nil, so can this method just become: 

```
return lhs.threadRecord.uniqueId == rhs.thread.threadRecord.uniqueId
```

And in that case, it would be nice to remove the inheritance of the NSObject machinery.

Reply:
I misread this - I realize we're checking the thread **and** the message, if any.

In that case:

```
if lhs.thread.threadRecord.uniqueId != rhs.thread.threadRecord.uniqueId {
     return false
}
return lhs.messageId == rhs.messageId
```

or

```
return lhs.thread.threadRecord.uniqueId == rhs.thread.threadRecord.uniqueId &&
lhs.messageId == rhs.messageId
```

(

Reply:
Done.